commit.go

```diff
@@ -183,6 +183,12 @@ func (b *Builder) addManifest(ctx context.Context, manifestName string, imageSpe
 		create = true
 		list = manifests.Create()
 	} else {
+		locker, err := manifests.LockerForImage(b.store, manifestList.ID())
```

Comment:
Nit: I would have preferred to move that into a `manifestList.{L,Unl}ock()` but it does the trick.